tools/parsecallstack.py

```
@@ -0,0 +1,131 @@
+                reglist += lst[-8:]
+
+    return reglist
+
+def get_stackvalue(filename):
+
+    stackvalue = []
+    first = 1
+    with open(filename, mode='r') as fl:
+        for line in fl:
+            lst = line.strip('\n').split(' ')
+            if "up_stackdump:" in lst:
+                if first == 1:
+                    first += 1
+                    # strip ":" of sp
+                    sp = lst[-9].strip(':')
+                    # The first item is the sp to restore the stack.
+                    stackvalue.append(sp)
+                stackvalue += lst[-8:]
+
+    return stackvalue
+
+def generate_cmm(cpu, regs, stackvalue):
+
+    dir = os.getcwd()
+    filename = dir + "\\callstack.cmm"
+
+    with open(filename, mode='w') as fl:
+        # Select the CPU and symbol.
+        fl.write("SYStem.CPU " + cpu + "\n")
+        fl.write("SYS.M UP\n")
+        fl.write("Data.LOAD *\n")
+        fl.write("\n")
+
+        # Set R0-R15.
+        for num in range(len(regs)):
+            fl.write("Register.Set R" + str(num) + " 0x" + regs[num] +'\n')
+        fl.write('\n')
+
+        # Recover the value in stack.
+        sp = int("0x" + stackvalue[0], 16)
+        for num in range(len(stackvalue) - 1):
+            address = hex(sp + num * 4)
+            value = stackvalue[num + 1]
+            fl.write("Data.Set ZSD:" + str(address) + " %LE %Long 0x"
+                + str(value) +'\n')
+        fl.write('\n')
+
+        # Show the call stack.
+        fl.write("data.view %sYmbol.long " + str(hex(sp)) + '\n')
+        fl.write("frame.view /Locals /Caller" +'\n')
+
+if __name__ == "__main__":
+    try:
+        args = parse_args()
+        filename = args.filename
+        cpu = args.cputype
+        if (os.path.isfile(filename)):
+            regs = get_regs(filename)
+            stackvalue = get_stackvalue(filename)
+            generate_cmm(cpu, regs, stackvalue)
+        else:
+            print("The file is not exist!")
```

Lets raise an exception on these errors so we have a non zero return code and also write to stderr or handle in a similar way if you dont want the exception information.